### DISCUSSION TITLE
How to force a parameter to be the last?

### DISCUSSION THREAD

**Discussion by giaphattram** [remkop/picocli#1996] [Q&A] [answered]
I have one parameter and a number of options. How to configure so that the parameter must take the last position?

**Comment by remkop** [comment, marked as answer]
There's no way in picocli to enforce that positional parameters need to follow the named options. 

So any validation will need to be done in the application. 

One idea is to get the [`ParseResult`](https://picocli.info/apidocs/picocli/CommandLine.ParseResult.html) object from the `CommandLine` object, and call the [matchedArgs](https://picocli.info/apidocs/picocli/CommandLine.ParseResult.html#matchedArgs--) or  [expandedArgs](https://picocli.info/apidocs/picocli/CommandLine.ParseResult.html#expandedArgs--) method on it to get the matched arguments or the original command line arguments. Then check in your application logic that the last element in the list was the positional parameter.

**Reply by giaphattram** [reply]
Thank you!
